library/src/commonMain/kotlin/com/mercury/sqkon/db/JsonPath.kt

```diff
@@ -117,20 +118,24 @@ class JsonPathBuilder<R : Any>
     }
 
     @OptIn(ExperimentalSerializationApi::class)
-    fun fieldNames(): List<String> {
+    fun fieldNames(element: Boolean = false): List<String> {
         return nodes().mapNotNull { it ->
             if (it.receiverDescriptor.isInline) return@mapNotNull null // Skip inline classes
             val prefix = if (it.propertyName.isNotBlank()) "." else ""
-            when (it.valueDescriptor.kind) {
-                StructureKind.LIST -> "$prefix${it.propertyName}[%]"
-                PolymorphicKind.SEALED -> "$prefix${it.propertyName}[1]"
-                else -> "$prefix${it.propertyName}"
+            if (element) {
+                when (it.valueDescriptor.kind) {
+                    StructureKind.LIST -> "$prefix${it.propertyName}[%]"
+                    PolymorphicKind.SEALED -> "$prefix${it.propertyName}[1]"
+                    else -> "$prefix${it.propertyName}"
+                }
+            } else {
+                "$prefix${it.propertyName}"
```

Comment:
The `element` arg is a hack. Sometimes we want the "outer" collection path, e.g. `$.property`, for example when comparing via `eq`:

```kotlin
TestObject::list eq listOf("one", "two", "three")
```

Other times we want the "inner" path, e.g. `$.property[%]`, for example when filtering by `contains`:

```kotlin
TestObject::list contains "one"
```

(Note that I added `contains`, which is the opposite of `inList`)

Reply:
Oh I think I see what you are saying, if the field is a list, inList doesn't make sense as thats checking the entire contents of that fields vs a list to list match